src/main/java/org/jitsi/protocol/xmpp/util/MediaSSRCMap.java

```diff
@@ -181,25 +231,18 @@ public MediaSSRCMap remove(MediaSSRCMap mapToRemove)
     }
 
     /**
-     * Returns shallow copy of this map. <tt>SourcePacketExtension</tt>
-     * instances are not copied, but referenced from both copy and this
-     * instance.
+     * Removes given SSRC from this map.
+     *
+     * @param media the media type of the SSRC to be removed.
+     * @param ssrc the <tt>SourcePacketExtension</tt> to be removed from this
+     *        <tt>MediaSSRCMap</tt>.
+     *
+     * @return <tt>true</tt> if the SSRC has been actually removed which means
+     *         that it was in the map before the operation took place.
      */
-    public MediaSSRCMap copyShallow()
+    public boolean remove(String media, SourcePacketExtension ssrc)
     {
-        Map<String, List<SourcePacketExtension>> mapCopy = new HashMap<>();
-
-        for (Map.Entry<String, List<SourcePacketExtension>> e
-                : ssrcs.entrySet())
-        {
-            String media = e.getKey();
-            List<SourcePacketExtension> listCopy
-                = new ArrayList<>(e.getValue());
-
-            mapCopy.put(media, listCopy);
-        }
-
-        return new MediaSSRCMap(mapCopy);
+        return getSSRCsForMedia(media).remove(ssrc);
```

Comment:
Oh yes, will change that to find the SSRC instance to be removed. There is a method for that already.